client/src/app/shared/http/auth_interceptor.service.ts

```diff
@@ -0,0 +1,66 @@
+import {HttpEvent, HttpHandler, HttpInterceptor, HttpRequest} from "@angular/common/http";
+import {BehaviorSubject, Observable} from "rxjs";
+import {Inject, Injectable} from "@angular/core";
+import {Session, SESSION, Token} from "../security/session.service";
+import {Authenticator, AUTHENTICATOR} from "../security/authenticator.service";
+import {catchError, filter, switchMap, take} from "rxjs/operators";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthInterceptor implements HttpInterceptor {
+  private isRefreshing = false;
+  private tokenSubject: BehaviorSubject<any> = new BehaviorSubject<any>(null);
+
+  constructor(@Inject(SESSION) private session: Session, @Inject(AUTHENTICATOR) private authenticator: Authenticator) {
+  }
+
+  private static addTokenHeader(req: HttpRequest<any>, token: Token): HttpRequest<any> {
+    return req.clone({
+      setHeaders: {
+        Authorization: `Bearer ${token.token}`
+      }
+    });
+  }
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    let authReq = req;
+    if (this.session.authenticated()) {
+      // @ts-ignore
+      authReq = AuthInterceptor.addTokenHeader(authReq, this.session.getToken());
+    }
+
+    return next.handle(authReq).pipe(catchError(error => {
+      if (!authReq.url.includes('/api/security/login') && error.status === 401) {
+        return this.handleJwtExpired(authReq, next);
+      }
+      return error;
```

Comment:
Le type retour de ```error``` est un any, par consequent il ne peut pas être un  `Observable<HttpEvent<any>>.`

Deux solutions : 

1. cast avec `as Observable<HttpEvent<any>>`
2. `return of(error)`